tests/integration_cases/test_raft.rs

```diff
@@ -2620,6 +2620,46 @@ fn test_leader_increase_next() {
     }
 }
 
+#[test]
+fn test_send_append_pagination_with_max_size_per_msg() {
```

Comment:
What's the purpose of this case? Maybe you can add a hook to `Network` to implement a similar case with the original PR.

Reply:
It ensures the  `hard_state.commit` matches with last committed entry index in `Ready` if the `committed_entries` overflow the `max_committed_size_per_ready`.